tuna/update_golden.py

```diff
@@ -150,57 +151,116 @@ def update_gold_entry(session, golden_table, gold_entry):
   return ret
 
 
-def merge_golden_entries(dbt, golden_v, entries):
+def sess_info(session):
+  """ get map for session id to arch / num_cu """
+  sess_map = {}
+  query = session.query(Session.id, Session.arch, Session.num_cu)
+  for entry in query.all():
+    sess_map[entry.id] = (entry.arch, entry.num_cu)
+
+  return sess_map
+
+
+def init_gold_entry(dbt, golden_v, config, solver, arch, num_cu):
+  """ initialize golden_table entry with key values """
+  gold_entry = dbt.golden_table()
+  #unique identifiers
+  gold_entry.golden_miopen_v = golden_v
+  gold_entry.config = config
+  gold_entry.solver = solver
+  gold_entry.arch = arch
+  gold_entry.num_cu = num_cu
+  return gold_entry
+
+
+def copy_gold_data(gold_entry, entry):
+  """ copy data fields for a golden_table entry """
+  gold_entry.valid = 1
+  gold_entry.session = entry.session
+
+  gold_entry.fdb_key = entry.fdb_key
+  gold_entry.params = entry.params
+  gold_entry.kernel_time = entry.kernel_time
+  gold_entry.workspace_sz = entry.workspace_sz
+  gold_entry.alg_lib = entry.alg_lib
+  gold_entry.opencl = entry.opencl
+
+  gold_entry.kernel_group = entry.kernel_group
+
+
+def merge_golden_entries(session, dbt, golden_v, entries, simple_copy=False):
   """! Retrieve fdb entries and populate golden table"""
+  sess_map = sess_info(session)
```

Comment:
If you need arch/num_cu from the session, since you already have dbt in this function you should be able to do:
dbt.session.arch and dbt.session.num_cu and not need to compose the session map

Reply:
items in entries could come from a set of gold_miopen_v from conv_golden, so they are not guaranteed to be all part of the same session. So dbt cannot store all the unique sessions.